automation/taskcluster/lib/artifacts.py

```diff
@@ -0,0 +1,22 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+import subprocess
+
+
+def from_gradle():
+    process = subprocess.Popen(["./gradlew", "--no-daemon", "printModules"], stdout=subprocess.PIPE)
+    (output, err) = process.communicate()
+    exit_code = process.wait()
+
+    if exit_code is not 0:
+        print "Gradle command returned error:", exit_code
+
+    tuples = re.findall('module: name=(.*) buildPath=(.*)', output, re.M)
+    return map(lambda (name, build_path): {
+        'name': name[1:],
+        'artifact': 'public/build/' + name[1:].replace('-', '.', 1) + '.maven.zip',
```

Comment:
Nit: Even though string concatenation (with `+`) is faster, `''.format()` is generally preferred when dealing with a handful of strings https://stackoverflow.com/questions/376461/string-concatenation-vs-string-substitution-in-python. It's a generally a matter of readability. 

We can also get rid of `.replace('-', '.', 1)`. The build and the beetmover task are the only consumers of this line (if I remember correctly). The reason we made it this way is because:
* artifacts.yml was a quick hack to get a-c deployed on maven.mozilla.org, a month and a half ago (https://github.com/mozilla-mobile/android-components/pull/803)
* some components got their name changed, in the meantime. 

This would become:

```py
# 
'artifact': 'public/build/{}.maven.zip'.format(name[1:]),
```

Reply:
Awesome, I wanted to avoid breaking changes, but if I've got your go-ahead, then I'd love to make this simpler!